mailpoet/lib/Util/Notices/SenderDomainAuthenticationNotices.php

```diff
@@ -0,0 +1,167 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Util\Notices;
+
+use MailPoet\Services\AuthorizedSenderDomainController;
+use MailPoet\Services\Bridge;
+use MailPoet\Settings\SettingsController;
+use MailPoet\Util\FreeDomains;
+use MailPoet\Util\Helpers;
+use MailPoet\Util\License\Features\Subscribers;
+use MailPoet\WP\Notice;
+
+class SenderDomainAuthenticationNotices {
+
+  const LOWER_LIMIT = 500;
+  const UPPER_LIMIT = 1000;
+
+  const FREE_MAIL_KB_URL = 'https://kb.mailpoet.com/article/259-your-from-address-cannot-be-yahoo-com-gmail-com-outlook-com';
+  const SPF_DKIM_DMARC_KB_URL = 'https://kb.mailpoet.com/article/295-spf-dkim-dmarc';
+
+  private SettingsController $settingsController;
+
+  private Subscribers $subscribersFeatures;
+
+  private FreeDomains $freeDomains;
+
+  private AuthorizedSenderDomainController $authorizedSenderDomainController;
+
+  private Bridge $bridge;
+
+  public function __construct(
+    SettingsController $settingsController,
+    Subscribers $subscribersFeatures,
+    FreeDomains $freeDomains,
+    AuthorizedSenderDomainController $authorizedEmailsController,
+    Bridge $bridge
+  ) {
+    $this->settingsController = $settingsController;
+    $this->subscribersFeatures = $subscribersFeatures;
+    $this->freeDomains = $freeDomains;
+    $this->authorizedSenderDomainController = $authorizedEmailsController;
+    $this->bridge = $bridge;
+  }
+
+  public function getDefaultFromAddress(): string {
+    return $this->settingsController->get('sender.address', '');
+  }
+
+  public function getDefaultFromDomain(): string {
+    return Helpers::extractEmailDomain($this->getDefaultFromAddress());
+  }
+
+  public function isFreeMailUser(): bool {
+    return $this->freeDomains->isEmailOnFreeDomain($this->getDefaultFromDomain());
+  }
+
+  public function init($shouldDisplay): ?Notice {
+    if (
+      !$shouldDisplay
+      || !$this->bridge->isMailpoetSendingServiceEnabled()
+      || in_array($this->getDefaultFromDomain(), $this->authorizedSenderDomainController->getFullyVerifiedSenderDomains(true))
+      || $this->isFreeMailUser() && $this->subscribersFeatures->getSubscribersCount() <= self::LOWER_LIMIT
+    ) {
+      return null;
+    }
+
+    return $this->display();
+  }
+
+  public function display(): Notice {
+    $contactCount = $this->subscribersFeatures->getSubscribersCount();
+    $isFreeMailUser = $this->isFreeMailUser();
+
+    $isErrorStyle = $contactCount > self::UPPER_LIMIT;
+
+    $noticeContent = $isFreeMailUser
+      ? $this->getNoticeContentForFreeMailUsers($contactCount)
+      : $this->getNoticeContentForBrandedDomainUsers(in_array($this->getDefaultFromDomain(), $this->authorizedSenderDomainController->getPartiallyVerifiedSenderDomains(true)), $contactCount);
+
```

Comment:
This is my mistake because the specs say that the notices are non dismissible but there is a close icon on the designs.
Could you please add the `$extraClasses` for that?

Reply:
Do we want them to stay dismissed? I'm guessing not but just want to confirm.